internal/services/securitycenter/security_center_server_vulnerability_assessment_vm_resource.go

```diff
@@ -0,0 +1,145 @@
+	computeValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+const (
+	computeProvider = "Microsoft.Compute"
+	vmType          = "virtualMachines"
+)
+
+func resourceServerVulnerabilityAssessmentVM() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceServerVulnerabilityAssessmentVMCreate,
+		Read:   resourceServerVulnerabilityAssessmentVMRead,
+		Delete: resourceServerVulnerabilityAssessmentVMDelete,
+
+		// TODO: replace this with an importer which validates the ID during import
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(5 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"virtual_machine_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.VirtualMachineID,
+			},
+		},
+	}
+}
+
+func resourceServerVulnerabilityAssessmentVMCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	virtualMachineId, err := computeParse.VirtualMachineID(d.Get("virtual_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	vulnerabilityAssessment, err := client.Get(ctx, virtualMachineId.ResourceGroup, computeProvider, vmType, virtualMachineId.Name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(vulnerabilityAssessment.Response) {
+			return fmt.Errorf("checking for presence of existing Advanced Threat Protection for %s: %+v", *virtualMachineId, err)
+		}
+	}
+
+	if vulnerabilityAssessment.ID != nil && *vulnerabilityAssessment.ID != "" {
+		return tf.ImportAsExistsError("azurerm_security_center_server_vulnerability_assessment_vm", *vulnerabilityAssessment.ID)
+	}
+
+	vulnerabilityAssessment, err = client.CreateOrUpdate(ctx, virtualMachineId.ResourceGroup, computeProvider, vmType, virtualMachineId.Name)
+	if err != nil {
+		return fmt.Errorf("create Server Vulnerability Assessment for %s: %+v", *virtualMachineId, err)
+	}
+
+	if vulnerabilityAssessment.ID != nil {
+		d.SetId(*vulnerabilityAssessment.ID)
+	}
+
+	// Take some time for resource to be visible after creating. Hence the retry logic
+	readError := resourceServerVulnerabilityAssessmentRead(d, meta)
+	for retry := 1; d.Id() == "" && retry < 11; retry++ {
+		time.Sleep(time.Duration(retry) * time.Second)
+		readError = resourceServerVulnerabilityAssessmentRead(d, meta)
+	}
+	return readError
```

Comment:
we can use a `WaitForState` function here to continually poll until the Vulnerability Assessment is returned from the API - which means we can instead return the Read function here?